### PR TITLE
Added the "Merge K sorted Arrays" problem

### PR DESCRIPTION
**Pull request by muskangarg12**
Added the solution to the problem named "Merge K sorted Arrays" using heap.

The problem statement is as follows :
"Given K sorted arrays arranged in the form of a matrix of size K*K. The task is to merge them into one sorted array."

### CODE REVIEW

File: Heap/Must Do/Merge K Sorted Arrays.cpp [hunk 1]
```diff
@@ -0,0 +1,44 @@
...
+// The merged list will be 
+// [1, 2, 3, 4, 5, 6, 7, 8, 9].
+
+// Approach: Heaps
+// The idea is to use Min Heap. This MinHeap based solution has the same time complexity which is O(NKlogK).
+// The process must start with creating a MinHeap and inserting the first element of all the k arrays to the heap.
+// Remove the root element of Minheap and put it in the output array and insert the next element from the array of removed element. 
+// To get the result the step must continue until there is no element in the MinHeap or the arrays left.
+
+int *mergeKArrays(int arr[][N], int k)
```
Comment (hmahajan99) [inline Heap/Must Do/Merge K Sorted Arrays.cpp]: Move the opening brace here

File: Heap/Must Do/Merge K Sorted Arrays.cpp [hunk 2]
```diff
@@ -0,0 +1,44 @@
...
+{
+    int *ans = new int[k*k];
+    priority_queue< pair<int, int>, vector<pair<int, int>>, greater<pair<int, int>> > q;
+    
+    int maxn[k];
+    for(int i=0;i<k;i++)
+    {
```
Comment (hmahajan99) [inline Heap/Must Do/Merge K Sorted Arrays.cpp]: Same here

File: Heap/Must Do/Merge K Sorted Arrays.cpp [hunk 3]
```diff
@@ -0,0 +1,44 @@
...
+        ans[i]=val.first;i++;
+        q.pop();
+        if(maxn[val.second]<k)
+        {
+            q.push({arr[val.second][maxn[val.second]],val.second});
+            maxn[val.second]++;
+        }
+    }
+    return ans;
+}
```
Comment (hmahajan99) [inline Heap/Must Do/Merge K Sorted Arrays.cpp:41]: LGTM!

**Review by hmahajan99 — CHANGES_REQUESTED** [review]
LGTM, just move the opening braces to the same line to maintain code style.

File: Heap/Must Do/Merge K Sorted Arrays.cpp (hunk 1 above)
Comment (muskangarg12) [reply Heap/Must Do/Merge K Sorted Arrays.cpp]: Done.

**Pull request merged**